student-store-ui/src/components/ProductCard/ProductCard.jsx

```diff
@@ -0,0 +1,63 @@
+import { Link } from "react-router-dom"
+
+import "./ProductCard.css"
+
+export default function ProductCard({ product, cart, setCart }) {
+  const handleIncrement = () => {
+    if (!cart || !cart.find(item => item.id === product.id)) {
+      let newItem = {
+        "id": product.id,
+        "name": product.name,
+        "price": product.price.toFixed(2),
+        "quantity": 1
+      }
+      setCart([...cart, newItem])
+    } else {
+      let updatedCart = [...cart]
+      let itemIndex = updatedCart.findIndex(item => item.id === product.id)
+      updatedCart[itemIndex].quantity += 1
+      setCart(updatedCart)
```

Comment:
Good approach to avoid mutating state directly! I recommend checking the exemplar code for another approach to expose yourself to different ways of solving the same problem.